File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/NullSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface NullSchemaValidator<T> {
    Void validate(Void arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
    T validateAndBox(Void arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/NullSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface NullSchemaValidator<T> {
    @Throws(ValidationException::class)
    fun validate(arg: Nothing?, configuration: SchemaConfiguration?): Nothing?

    @Throws(ValidationException::class)
    fun validateAndBox(arg: Nothing?, configuration: SchemaConfiguration?): T
}